src/opm/io/eclipse/ESmry.cpp

```diff
@@ -241,65 +252,70 @@ ESmry::ESmry(const std::string &filename, bool loadBaseRunData) :
         nJ = dimens[2];
         nK = dimens[3];
 
+        nParamsSpecFile[specInd] = dimens[0];
+
         const std::vector<std::string> keywords = smspec.get<std::string>("KEYWORDS");
         const std::vector<std::string> wgnames = smspec.get<std::string>("WGNAMES");
         const std::vector<int> nums = smspec.get<int>("NUMS");
 
-        const std::vector<int> tmpVect(keywords.size(), -1);
-        arrayInd[n]=tmpVect;
-
-        std::set<std::string>::iterator it;
-
         for (size_t i=0; i < keywords.size(); i++) {
             const std::string keyw = makeKeyString(keywords[i], wgnames[i], nums[i]);
-            it = std::find(keywList.begin(), keywList.end(), keyw);
 
-            if (it != keywList.end()){
-                arrayInd[n][i] = distance(keywList.begin(), it);
-            }
+            if (keywList.find(keyw) != keywList.end())
+                arrayPos[specInd][keyIndex[keyw]]=i;
         }
 
-        n--;
+        specInd--;
     }
 
-    // param array used to stor data for the object, defined in the private section of the class
-    param.assign(keywList.size(), {});
-
     int fromReportStepNumber = 0;
     int toReportStepNumber;
-
-    float time = 0.0;
     int step = 0;
+    specInd = nSpecFiles - 1;
+    nVect = keywList.size();
+
+    int index = 0;
+    for (const auto& keyw : keywList){
+        keyword.push_back(keyw);
+        keyword_index[keyw] = index++;
+    }
 
-    n = nFiles - 1;
+    vectorData.reserve(nVect);
+    vectorLoaded.reserve(nVect);
+
+    for (size_t n = 0; n < nVect; n ++){
+        vectorData.push_back({});
+        vectorLoaded.push_back(false);
+    }
 
-    while (n >= 0){
+    int dataFileIndex = -1;
+
+    while (specInd >= 0){
 
         int reportStepNumber = fromReportStepNumber;
 
-        if (n > 0) {
-            auto rstFrom = smryArray[n-1];
+        if (specInd > 0) {
+            auto rstFrom = smryArray[specInd-1];
             toReportStepNumber = std::get<1>(rstFrom);
         } else {
             toReportStepNumber = std::numeric_limits<int>::max();
         }
 
-        Opm::filesystem::path smspecFile(std::get<0>(smryArray[n]));
+        Opm::filesystem::path smspecFile(std::get<0>(smryArray[specInd]));
         rootName = smspecFile.parent_path() / smspecFile.stem();
 
-
         // check if multiple or unified result files should be used
         // to import data, no information in smspec file regarding this
         // if both unified and non-unified files exists, will use most recent based on
         // time stamp
 
         Opm::filesystem::path unsmryFile = rootName;
 
-        formattedVect[n] ? unsmryFile += ".FUNSMRY" : unsmryFile += ".UNSMRY";
+        formattedFiles[specInd] ? unsmryFile += ".FUNSMRY" : unsmryFile += ".UNSMRY";
```

Comment:
I think I would rather write this in the more conventional form
```C++
unsmryFile += formattedFiles[specInd] ? ".FUNSMRY" : ".UNSMRY";
```
At least this is what most developers are used to seeing.